### ISSUE TITLE
improve logging

### ISSUE THREAD

**Issue by marcoieni** [release-plz/release-plz#1739]
## Motivations
Right now logging is very verbose. It was useful at the beginning of release-plz to troubleshoot bugs, but now not anymore. We should keep the traces with `--verbose` cli arg or similar

## Solution

Much simpler logging, with the following requirements:
- tell when publishing something (e.g. by showing cargo commands)
- tell that cargo-semver-checks is checking something

**Comment by EleisonC** [comment]
Hey @MarcoIeni can I take this on?

**Comment by MarcoIeni** [comment]
Sure, but I might heavily edit your PR based on my taste. If that's ok with you, go ahead :)

**Comment by EleisonC** [comment]
I get to learn from the best. No Worries

**Comment by EleisonC** [comment]
Hey @MarcoIeni I have a few questions for clarity around my task.  
- Ideally, my target is to look at places where logging is verbose (What determines verbose) ie in the

**Comment by EleisonC** [comment]
Hey @MarcoIeni I have a few questions for clarity around my task.  
- What constitutes verbosity?
- Ideally, my target is to look at places where logging is verbose ie in the 
get package diff function.
**The current info log in file path** -crates/release_plz_core/src/next_ver.rs
*log below in info! when verbose is true*
` info!("{}: the local package has already a different version with respect to the registry package, so release-plz will not update it", package.name);`
**Change the above info! when verbose is false **  When ver
` info!("{}: Local version differs from registry version; skipping update", package.name);`

Would that be it?

**Comment by MarcoIeni** [comment]
No, with verbose I mean that the traces.
Only the info line should be present, not the file, line number, etc.

**Comment by EleisonC** [comment]
![Screenshot 2024-11-18 at 21 13 57](https://github.com/user-attachments/assets/b46556cb-b3d9-400b-b734-1b2ca7a9de85)
Hey @MarcoIeni I also noticed that errors display twice like in the screenshot, is this something you are aware of? if not I don't mind  looking into it

**Comment by MarcoIeni** [comment]
Yeah, I'm aware of this. Errors are printed twice: the first time it's logged and at the end of the program.

It happens sometimes that people don't look at the full log. They just look at the final lines of the CI run logs.

I know it's ugly but at least it's effective 😅

**Comment by EleisonC** [comment]
> It happens sometimes that people don't look at the full log.

You are not wrong on this one 😅.

**Comment by joshka** [comment]
There's a few things which I'd suggest improving:

1. Every line that logs should have a timestamp as this helps understand which parts are fast and slow. It's meaningless when everything is working fine, but if you're looking at the logs, then that's not the case. right now, there's a few parts of release-plz that don't properly log the timestamps (in the cargo download parts).
2. When a time consuming task is about to be undertaking, log before the task. Right now after the download part there's a large gap of time which has no obvious reason from the logs.
3. Colorizing the entire line is meaningless. Just color the severity.
4. Where there's problems, log the details of the things which are causing the problems. Right now errors / log message don't carry enough context to identify what error is being caused. E.g.![Image](https://github.com/user-attachments/assets/1d8287b7-69f7-4688-b2fd-d5382db74277)
